tests/keyboard/scancode/test_unicode_to_scancode.cpp

```diff
@@ -0,0 +1,301 @@
+#include "test_only/test_framework/redemption_unit_tests.hpp"
+
+#include "keyboard/scancode/unicode_to_scancode.hpp"
+
+namespace
+{
+    using namespace scancode;
+
+    Scancode16bits_t get_and_pop_elem(ScancodeSeq_t& scancode_seq)
+    {
+        Scancode16bits_t scancode = scancode_seq.front();
+
+        scancode_seq.pop();
+        return scancode;
+    }
+
+
+    class FrenchLocaleFixture
+    {
+      protected :
+        static constexpr int _LCID = 0x0000040c;
+    };
+}
+
+
+RED_FIXTURE_TEST_CASE(unicode_to_scancode__simple_unicode_no_key_combination,
+                      FrenchLocaleFixture)
+{
+    {
+        constexpr std::uint16_t UNICODE = 0x0061; // 'a'
+        ScancodeSeq_t scancode_seq = unicode_to_scancode(_LCID, UNICODE);
+
+        RED_REQUIRE_EQUAL(scancode_seq.size(), 1);
+
+        
+        Scancode16bits_t pressed_scancode = get_and_pop_elem(scancode_seq);
+        
+        RED_CHECK_EQUAL(pressed_scancode, 0x0010); // 'A' scancode
+    }
```

Comment:
```cpp
// see jsclient/tests/test_scancodes.cpp for a custom output (ex: hexadecimal rendering)
// @{
using U16CArray = uint16_t[];
#define U16A(...) make_array_view(U16CArray{__VA_ARGS__})
// @}

// with a left operand as std::vector or array_view
RED_CHECK(unicode_to_scancode(_LCID, 'a') == U16A('A'/*, etc*/));
```

Reply:
Isn't it dangerous to use U16A macro? I mean, you create a array_view from U16CArray array which will be a temporary and by copying the pointer in array_view and U16CArray array will be deleted from stack after outgoing of make_array_view() call

By making a similar test in parallel, it compiles with clang++ but not with g++ with the following error : 

> test.cpp:10:19: error: taking address of temporary array